sql/logictest: TestSqlLiteLogic failed

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1897785&tab=buildLog) on [master@0e16cc15f139b816b8e46fe6571691a8ec0e6937](https://github.com/cockroachdb/cockroach/commits/0e16cc15f139b816b8e46fe6571691a8ec0e6937):

```
goroutine 2370804 [select, 5 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).initRunners.func1(0x43a87e0, 0xc2c1045f20)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:100 +0x13f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc096442ec0, 0xc120d4b050, 0xc096442eb0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x13e
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xa8

goroutine 2371735 [chan receive, 5 minutes]:
github.com/cockroachdb/cockroach/pkg/gossip.(*server).start.func3(0x43a87e0, 0xc18a553d70)
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/server.go:383 +0x5a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc0ac578e70, 0xc120d4b050, 0xc06ca49d20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x13e
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xa8

goroutine 2477747 [select, 5 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).initRunners.func1(0x43a87e0, 0xc2b05c34a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:100 +0x13f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc3d56b3bf0, 0xc3d6512510, 0xc3d56b3be0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x13e
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xa8

goroutine 2501716 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:310
sync.runtime_notifyListWait(0xc5f5af5dd0, 0x842)
	/usr/local/go/src/runtime/sema.go:510 +0xf8
sync.(*Cond).Wait(0xc5f5af5dc0)
	/usr/local/go/src/sync/cond.go:56 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).worker(0xc5e8b01d00, 0x43a87e0, 0xc4079c42a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:197 +0xaa
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2(0x43a87e0, 0xc4079c42a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:166 +0x3e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc33d1f46c0, 0xc5fbde30e0, 0xc33d1f46b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x13e
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xa8

goroutine 2371744 [select, 5 minutes]:
github.com/cockroachdb/cockroach/pkg/internal/client/requestbatcher.(*RequestBatcher).run(0xc3abcf9440, 0x43a8760, 0xc000074120)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/requestbatcher/batcher.go:464 +0x390
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc120d4b050, 0x43a8760, 0xc000074120, 0xc309721920, 0x28, 0x0, 0x0, 0xc0ac5793d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:323 +0x152
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:318 +0x131
FAIL	github.com/cockroachdb/cockroach/pkg/sql/logictest	730.712s
=== RUN   TestSqlLiteLogic/local/select4.test
=== PAUSE TestSqlLiteLogic/local/select4.test

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1899749&tab=buildLog) on [master@d620f6242ad43481e61a6af19416733cf05233a4](https://github.com/cockroachdb/cockroach/commits/d620f6242ad43481e61a6af19416733cf05233a4):

Fatal error:
```
panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM _ WHERE (((_ > _) AND (_ <= _)) AND ((_ < _))) OR ((_ IS _) AND (_ = _)) ORDER BY _ DESC; caused by non-values node passed as fixed value to zigzag join

```

Stack:
```
goroutine 3911005 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc020db2000, 0x43ab200, 0xc0031ef240, 0x35dc8a0, 0x42e18a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc020db2000, 0x43ab200, 0xc0031ef240)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x35dc8a0, 0x42e18a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build.func1(0xc0518fe510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:117 +0xae
panic(0x35dc8a0, 0x42e18a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*execFactory).ConstructZigzagJoin(0xc07607ddc0, 0x4429a80, 0xc09080bb00, 0x44121a0, 0xc013ac33d0, 0x4429a80, 0xc09080bb00, 0x44121a0, 0xc013ac34c0, 0xc06608fdec, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:759 +0x870
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildZigzagJoin(0xc04360bef0, 0xc02257b680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fc4f7aaed37d8e6, 0xc0518fddf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1419 +0xd2e
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc04360bef0, 0x44195e0, 0xc02257b680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218 +0x8fc
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildProject(0xc04360bef0, 0xc0a2c0daa0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x393eaa0, 0x14c7c8aff1d8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:552 +0x89
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc04360bef0, 0x44188c0, 0xc0a2c0daa0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66cf9b, 0xc00004e000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:197 +0x7ea
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildSort(0xc04360bef0, 0xc08f1f5080, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0518fe2f0, 0xd8fc89)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1219 +0x79
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc04360bef0, 0x4418fa0, 0xc08f1f5080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0518fe460)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:209 +0x1692
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build(0xc04360bef0, 0x43cc940, 0xc08f1f5080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:131 +0x12d
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).Build(0xc04360bef0, 0xc04360bef0, 0xc00e04b020, 0xc01e848d00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:100 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc020db23d0, 0x43ab2c0, 0xc00e04b020, 0xc0518fe650, 0x72adb6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:178 +0x1c6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc020db2000, 0x43ab2c0, 0xc00e04b020, 0xc020db23d0, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:801 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc020db2000, 0x43ab2c0, 0xc00e04b020, 0xc020db23d0, 0x14c7c8b9c200, 0xc05e7fb380, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:691 +0xf9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc020db2000, 0x43ab2c0, 0xc00e04b020, 0x43c4880, 0xc083aeb0e0, 0xc01a9a8e9e, 0x7e, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:481 +0xb06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc020db2000, 0x43ab2c0, 0xc00e04b020, 0x43c4880, 0xc083aeb0e0, 0xc01a9a8e9e, 0x7e, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:96 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc020db2000, 0x43ab200, 0xc02a543b80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1368 +0x1aa8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc020db2000, 0x43ab200, 0xc0031ef240, 0xc0b3ccb300, 0x5400, 0x15000, 0xc0b3ccb398, 0xc09aab3330, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc02b76ab00, 0x43ab200, 0xc0031ef240, 0xc020db2000, 0x5400, 0x15000, 0xc0b3ccb398, 0xc09aab3330, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0850b48f2, 0xc07360a880, 0x43ab200, 0xc0031ef240, 0xc09aab3330, 0xc02b76ab00, 0xc05e7fad00, 0x43cd1c0, 0xc06f295800, 0xc075726d80, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:595 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:527 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 3623 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7256 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3829 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2757 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 906 statements/queries
rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1), FAMILY (col2), FAMILY (col4), FAMILY (col5), FAMILY (col3, pk), FAMILY (col0));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3693 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1493 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 3700 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1664 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7303 statements/queries
rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (pk), FAMILY (col1), FAMILY (col4), FAMILY (col2, col3), FAMILY (col0), FAMILY (col5));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3920 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2847 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1012 statements/queries
rewrote:
CREATE TABLE tab3 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col5), FAMILY (col1), FAMILY (col3, col4), FAMILY (col0), FAMILY (col2), FAMILY (pk));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3771 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1584 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 3771 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1756 statements/queries
rewrote:
CREATE TABLE tab4 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col0), FAMILY (pk), FAMILY (col4, col3, col5), FAMILY (col1), FAMILY (col2));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7371 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3998 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2924 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1029 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3841 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1675 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 3852 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1848 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7428 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 4067 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2999 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1273 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1758 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3922 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1938 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 3921 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7486 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1900797&tab=buildLog) on [master@60c9e055e970bd7f150ebcfad266929b2638d635](https://github.com/cockroachdb/cockroach/commits/60c9e055e970bd7f150ebcfad266929b2638d635):

Fatal error:
```
panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM _ WHERE (((_ IS _) AND (_ < _)) AND (_ >= _)) OR ((_ IS _) AND (_ IS _)) ORDER BY _ DESC; caused by non-values node passed as fixed value to zigzag join

```

Stack:
```
goroutine 2705945 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc002dd9000, 0x43b8160, 0xc086458400, 0x35e4c80, 0x42ee380)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc002dd9000, 0x43b8160, 0xc086458400)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x35e4c80, 0x42ee380)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build.func1(0xc015102510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:117 +0xae
panic(0x35e4c80, 0x42ee380)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*execFactory).ConstructZigzagJoin(0xc00863d0f0, 0x4436b40, 0xc05cfb6780, 0x441f1c0, 0xc0c45c98b0, 0x4436b40, 0xc05cfb6780, 0x441f1c0, 0xc0c45c9950, 0xc03f4ce93c, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:775 +0x870
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildZigzagJoin(0xc0d7c2d6b0, 0xc01e91a000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fbcfbf054f3b96c, 0xc015101df0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1412 +0xd2e
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc0d7c2d6b0, 0x44266a0, 0xc01e91a000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4436b40, 0x7)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218 +0x8fc
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildProject(0xc0d7c2d6b0, 0xc0115b95a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3947a60, 0x14d736c3f1b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:552 +0x89
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc0d7c2d6b0, 0x4425980, 0xc0115b95a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66cf9b, 0xc00004e000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:197 +0x7ea
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildSort(0xc0d7c2d6b0, 0xc033506120, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0151022f0, 0xd8fc89)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1212 +0x79
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc0d7c2d6b0, 0x4426060, 0xc033506120, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc015102460)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:209 +0x16a9
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build(0xc0d7c2d6b0, 0x43d98e0, 0xc033506120, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:131 +0x12d
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).Build(0xc0d7c2d6b0, 0xc0d7c2d6b0, 0xc0a2c93860, 0xc0313edd40, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:100 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc002dd93d0, 0x43b8220, 0xc0a2c93860, 0xc015102650, 0x72adb6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:178 +0x1c6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc002dd9000, 0x43b8220, 0xc0a2c93860, 0xc002dd93d0, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:801 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc002dd9000, 0x43b8220, 0xc0a2c93860, 0xc002dd93d0, 0x14d738244348, 0xc0608f4f80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:691 +0xf9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc002dd9000, 0x43b8220, 0xc0a2c93860, 0x43d1820, 0xc00ce6bae0, 0xc000effb32, 0x78, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:481 +0xb06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc002dd9000, 0x43b8220, 0xc0a2c93860, 0x43d1820, 0xc00ce6bae0, 0xc000effb32, 0x78, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:96 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc002dd9000, 0x43b8160, 0xc023751b40, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1368 +0x1aa8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc002dd9000, 0x43b8160, 0xc086458400, 0xc01ce5a100, 0x5400, 0x15000, 0xc01ce5a198, 0xc01d627320, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc000e3e580, 0x43b8160, 0xc086458400, 0xc002dd9000, 0x5400, 0x15000, 0xc01ce5a198, 0xc01d627320, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc02267aefd, 0xc06bdf7960, 0x43b8160, 0xc086458400, 0xc01d627320, 0xc000e3e580, 0xc0608f4900, 0x43da160, 0xc0621f24e0, 0xc017eb3560, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:595 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:527 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestSqlLiteLogic/local/slt_good_33.test#01
=== PAUSE TestSqlLiteLogic/local/slt_good_33.test#01
=== CONT  TestSqlLiteLogic/local/slt_good_33.test#01
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 4152 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_8.test: 889 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1902132&tab=buildLog) on [master@a20a8811ee6abfe3754220e893ed383afbab21c9](https://github.com/cockroachdb/cockroach/commits/a20a8811ee6abfe3754220e893ed383afbab21c9):

Fatal error:
```
panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM _ WHERE (((_ > _) AND (_ <= _)) AND ((_ < _))) OR ((_ IS _) AND (_ = _)) ORDER BY _ DESC; caused by non-values node passed as fixed value to zigzag join

```

Stack:
```
goroutine 3718091 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc0838af000, 0x43d7f40, 0xc034874540, 0x35fae60, 0x430d9d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc0838af000, 0x43d7f40, 0xc034874540)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x35fae60, 0x430d9d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build.func1(0xc04b21a510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:117 +0xae
panic(0x35fae60, 0x430d9d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*execFactory).ConstructZigzagJoin(0xc03c2c2100, 0x4456b60, 0xc0952befc0, 0x443f200, 0xc0364deef0, 0x4456b60, 0xc0952befc0, 0x443f200, 0xc0364defe0, 0xc0b42e2f14, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:791 +0x870
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildZigzagJoin(0xc08646fef0, 0xc00c399200, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fc4f7aaed37d8e6, 0xc04b219df0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1469 +0xd2e
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc08646fef0, 0x4446780, 0xc00c399200, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4456b60, 0x7)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:221 +0x906
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildProject(0xc08646fef0, 0xc05afac920, 0x0, 0x0, 0x0, 0x0, 0x0, 0x395fc20, 0x14e918c8c2f8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:555 +0x89
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc08646fef0, 0x4445a60, 0xc05afac920, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66cf9b, 0xc00004e000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:197 +0x7f4
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildSort(0xc08646fef0, 0xc074560ae0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc04b21a2f0, 0xd8fc89)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1215 +0x79
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc08646fef0, 0x4446140, 0xc074560ae0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc04b21a460)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:209 +0x1724
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build(0xc08646fef0, 0x43f97c0, 0xc074560ae0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:131 +0x12d
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).Build(0xc08646fef0, 0xc08646fef0, 0xc004d4f320, 0xc01a8c56c0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:100 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc0838af3d0, 0x43d8000, 0xc004d4f320, 0xc04b21a650, 0x72adb6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:178 +0x1c6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc0838af000, 0x43d8000, 0xc004d4f320, 0xc0838af3d0, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:801 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0838af000, 0x43d8000, 0xc004d4f320, 0xc0838af3d0, 0x14e91a2f0260, 0xc086e81c80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:691 +0xf9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc0838af000, 0x43d8000, 0xc004d4f320, 0x43f1640, 0xc049925a40, 0xc0009bce9e, 0x7e, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:481 +0xb06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0838af000, 0x43d8000, 0xc004d4f320, 0x43f1640, 0xc049925a40, 0xc0009bce9e, 0x7e, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:96 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0838af000, 0x43d7f40, 0xc02f4bcd40, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1368 +0x1aa8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0838af000, 0x43d7f40, 0xc034874540, 0xc074327780, 0x5400, 0x15000, 0xc074327818, 0xc087ab7480, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc05bde4100, 0x43d7f40, 0xc034874540, 0xc0838af000, 0x5400, 0x15000, 0xc074327818, 0xc087ab7480, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc074a0706d, 0xc0232e20a0, 0x43d7f40, 0xc034874540, 0xc087ab7480, 0xc05bde4100, 0xc086e81600, 0x43fa040, 0xc010df0ae0, 0xc02870b560, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:595 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:527 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```

--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 2796 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 2482 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 2955 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1046 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1489 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 6926 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 866 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2309 statements/queries
rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col4), FAMILY (col2), FAMILY (pk, col3), FAMILY (col5), FAMILY (col0, col1));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 2896 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 2565 statements/queries
rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1, col5, col0, col4, pk), FAMILY (col2, col3));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3032 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1146 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1581 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 6981 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1011 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2398 statements/queries
rewrote:
CREATE TABLE tab3 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1), FAMILY (pk), FAMILY (col4, col2, col0), FAMILY (col3, col5));


rewrote:
CREATE TABLE tab4 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (pk, col5), FAMILY (col0), FAMILY (col1), FAMILY (col4), FAMILY (col2, col3));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 2979 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 2644 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3115 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1230 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1665 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7037 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1056 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2479 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3059 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 2726 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3185 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 1321 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 1756 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test: 1284 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 7096 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 2562 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_5.test: 3140 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1904531&tab=buildLog) on [master@c3bb2bce3d69d0ceef20797d856a6ac107a3ef47](https://github.com/cockroachdb/cockroach/commits/c3bb2bce3d69d0ceef20797d856a6ac107a3ef47):

Fatal error:
```
panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM _ WHERE ((((_ >= _) AND (_ IN (_, _, __more3__))) AND (((_ >= _)))) OR (_ = _)) OR (((_ IN (_,)) AND (_ IS _)) OR ((_ <= _) AND (_ = _))); caused by non-values node passed as fixed value to zigzag join

```

Stack:
```
goroutine 85065 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00f02a000, 0x43f5780, 0xc01e153780, 0x360bfa0, 0x432ae50)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00f02a000, 0x43f5780, 0xc01e153780)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x360bfa0, 0x432ae50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build.func1(0xc029560510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:117 +0xae
panic(0x360bfa0, 0x432ae50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*execFactory).ConstructZigzagJoin(0xc015bc4148, 0x4474520, 0xc002b218c0, 0x445cb00, 0xc018ee2720, 0x4474520, 0xc002b218c0, 0x445cb00, 0xc018ee27c0, 0xc0002a8a58, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:791 +0x870
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildZigzagJoin(0xc008db4bb0, 0xc009a901a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0080eb680, 0xc007564200)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1469 +0xd2e
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc008db4bb0, 0x4464080, 0xc009a901a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:221 +0x906
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildLookupJoin(0xc008db4bb0, 0xc02093cc80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1297 +0x85
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc008db4bb0, 0x4462e60, 0xc02093cc80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:215 +0xe4c
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildSetOp(0xc008db4bb0, 0x4463ae0, 0xc02093cba0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66d4d18, 0x66cf9b)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1121 +0x183
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc008db4bb0, 0x4463ae0, 0xc02093cba0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66d4d18, 0xc02955fc28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:304 +0x29a
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildGroupByInput(0xc008db4bb0, 0x4462280, 0xc0213b3dd0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2b6c53b, 0xc00f02abe8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1058 +0xcd
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildDistinct(0xc008db4bb0, 0x4462280, 0xc0213b3dd0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc006a8cae8, 0xc029560080)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1017 +0xf2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc008db4bb0, 0x4462280, 0xc0213b3dd0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4416e00, 0xc029560080)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:203 +0xe08
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildProject(0xc008db4bb0, 0xc02093c6a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0295602f0, 0xd8fc89)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:555 +0x89
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc008db4bb0, 0x4463360, 0xc02093c6a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc029560460)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:197 +0x7f4
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build(0xc008db4bb0, 0x4416ac0, 0xc02093c6a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:131 +0x12d
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).Build(0xc008db4bb0, 0xc008db4bb0, 0xc01c6f9560, 0xc00c4104e0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:100 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc00f02a3d0, 0x43f5840, 0xc01c6f9560, 0xc029560650, 0x72adb6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:178 +0x1c6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc00f02a000, 0x43f5840, 0xc01c6f9560, 0xc00f02a3d0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:801 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00f02a000, 0x43f5840, 0xc01c6f9560, 0xc00f02a3d0, 0x14d870d16968, 0xc019200f80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:691 +0xf9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00f02a000, 0x43f5840, 0xc01c6f9560, 0x440ee80, 0xc00e176550, 0xc028318ea4, 0xba, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:481 +0xb06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00f02a000, 0x43f5840, 0xc01c6f9560, 0x440ee80, 0xc00e176550, 0xc028318ea4, 0xba, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:96 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00f02a000, 0x43f5780, 0xc02b348980, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1368 +0x1aa8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00f02a000, 0x43f5780, 0xc01e153780, 0xc015871c00, 0x5400, 0x15000, 0xc015871c98, 0xc006b443c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00ac36b00, 0x43f5780, 0xc01e153780, 0xc00f02a000, 0x5400, 0x15000, 0xc015871c98, 0xc006b443c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc01cd6e202, 0xc016192860, 0x43f5780, 0xc01e153780, 0xc006b443c0, 0xc00ac36b00, 0xc019200900, 0x4417880, 0xc0062ec930, 0xc00cd9e660, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:595 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:527 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 3921 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3670 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 3987 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 6982 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4091 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 5891 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3519 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 3699 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 3934 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3712 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 3992 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 7092 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4199 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 5953 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3580 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 3858 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3731 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4042 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4109 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4358 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 7224 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 6031 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 4112 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3604 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4051 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3760 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4136 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4465 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 7230 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 6240 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 4271 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3706 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4152 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3785 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4251 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4585 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 7298 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 6324 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3824 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 4326 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3874 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4208 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4366 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 4752 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 7321 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 6399 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3956 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 4336 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4234 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 4151 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1907051&tab=buildLog) on [master@41a361bef43ad9653792219cb74576c1bfa80b99](https://github.com/cockroachdb/cockroach/commits/41a361bef43ad9653792219cb74576c1bfa80b99):

Fatal error:
```
panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: non-values node passed as fixed value to zigzag join [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM _ WHERE ((((_ >= _) AND (_ IN (_, _, __more3__))) AND (((_ >= _)))) OR (_ = _)) OR (((_ IN (_,)) AND (_ IS _)) OR ((_ <= _) AND (_ = _))); caused by non-values node passed as fixed value to zigzag join

```

Stack:
```
goroutine 327494 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc005d1f000, 0x43ed560, 0xc026aebdc0, 0x3604880, 0x4322920)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:811 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc005d1f000, 0x43ed560, 0xc026aebdc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:475 +0x61
panic(0x3604880, 0x4322920)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build.func1(0xc029e98510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:117 +0xae
panic(0x3604880, 0x4322920)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*execFactory).ConstructZigzagJoin(0xc00d3e06c8, 0x446c2c0, 0xc014b60240, 0x44547a0, 0xc010428f40, 0x446c2c0, 0xc014b60240, 0x44547a0, 0xc010428fe0, 0xc022c19570, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:791 +0x870
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildZigzagJoin(0xc00c258630, 0xc0100861a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc01533a480, 0xc010083f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1469 +0xd2e
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc00c258630, 0x445bd20, 0xc0100861a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:221 +0x906
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildLookupJoin(0xc00c258630, 0xc00ca25900, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1297 +0x85
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc00c258630, 0x445ab00, 0xc00ca25900, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:215 +0xe4c
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildSetOp(0xc00c258630, 0x445b780, 0xc00ca250a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66cce50, 0x66d29b)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1121 +0x183
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc00c258630, 0x445b780, 0xc00ca250a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66cce50, 0xc029e97c28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:304 +0x29a
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildGroupByInput(0xc00c258630, 0x4459f20, 0xc010af5710, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2b634fb, 0xc005d1fbe8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1058 +0xcd
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildDistinct(0xc00c258630, 0x4459f20, 0xc010af5710, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0239388a8, 0xc029e98080)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1017 +0xf2
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc00c258630, 0x4459f20, 0xc010af5710, 0x0, 0x0, 0x0, 0x0, 0x0, 0x440ebe0, 0xc029e98080)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:203 +0xe08
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildProject(0xc00c258630, 0xc00ca24e20, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc029e982f0, 0xd8ff89)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:555 +0x89
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).buildRelational(0xc00c258630, 0x445b000, 0xc00ca24e20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc029e98460)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:197 +0x7f4
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).build(0xc00c258630, 0x440e8a0, 0xc00ca24e20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:131 +0x12d
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder.(*Builder).Build(0xc00c258630, 0xc00c258630, 0xc012f93260, 0xc0142ad6c0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:100 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc005d1f3d0, 0x43ed620, 0xc012f93260, 0xc029e98650, 0x72b0b6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:178 +0x1c6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc005d1f000, 0x43ed620, 0xc012f93260, 0xc005d1f3d0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:801 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc005d1f000, 0x43ed620, 0xc012f93260, 0xc005d1f3d0, 0x14adbe894a68, 0xc00e930f80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:691 +0xf9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc005d1f000, 0x43ed620, 0xc012f93260, 0x4406c60, 0xc0119ef630, 0xc01aed1ea4, 0xba, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:481 +0xb06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc005d1f000, 0x43ed620, 0xc012f93260, 0x4406c60, 0xc0119ef630, 0xc01aed1ea4, 0xba, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:96 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc005d1f000, 0x43ed560, 0xc004eab4c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1369 +0x1aa8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc005d1f000, 0x43ed560, 0xc026aebdc0, 0xc008aed780, 0x5400, 0x15000, 0xc008aed818, 0xc015ab7980, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1298 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc004095600, 0x43ed560, 0xc026aebdc0, 0xc005d1f000, 0x5400, 0x15000, 0xc008aed818, 0xc015ab7980, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:477 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc00695b14d, 0xc01efe42e0, 0x43ed560, 0xc026aebdc0, 0xc015ab7980, 0xc004095600, 0xc00e930900, 0x440f660, 0xc01d80a090, 0xc0091a81e0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:595 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:527 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```

--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4302 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 2589 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 2883 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8727 statements/queries
rewrote:
CREATE TABLE tab0 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col3, col0, col2), FAMILY (col1), FAMILY (pk), FAMILY (col5), FAMILY (col4));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4212 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3387 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 3618 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_0.test: 6966 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4427 statements/queries
rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col5, col1, col3, col0, col2, pk, col4));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 2606 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3025 statements/queries
rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (pk), FAMILY (col1), FAMILY (col4), FAMILY (col5, col2), FAMILY (col3), FAMILY (col0));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8773 statements/queries
rewrote:
CREATE TABLE tab3 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1, col4), FAMILY (col0), FAMILY (col3), FAMILY (col5), FAMILY (pk), FAMILY (col2));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4350 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3414 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 3757 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4521 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_0.test: 7044 statements/queries
rewrote:
CREATE TABLE tab4 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1), FAMILY (col5), FAMILY (pk), FAMILY (col0, col2), FAMILY (col3, col4));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 2630 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test: 3111 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8871 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 4478 statements/queries
rewrote:
CREATE TABLE tab0 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1, col3), FAMILY (col0), FAMILY (col4, col2), FAMILY (pk), FAMILY (col5));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 3997 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3481 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_4.test: 4631 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_0.test: 7187 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/zigzag_join
+++ b/pkg/sql/logictest/testdata/logic_test/zigzag_join
@@ -59,6 +59,11 @@ CREATE INDEX a_idx_2 ON a(a);
 CREATE INDEX b_idx_2 ON a(b);
 SELECT n,a,b FROM a WHERE a = 4 AND b = 1;
 
+# Regression test for 48003 ("non-values node passed as fixed value to zigzag
+# join" error).
+statement ok
+SELECT n FROM a WHERE b = 1 AND (((a < 1) AND (a > 1)) OR (a >= 2 AND a <= 2))
+
 # ------------------------------------------------------------------------------
 # Zigzag join tests on inverted indexes.
 # ------------------------------------------------------------------------------

--- a/pkg/sql/opt/constraint/constraint_set.go
+++ b/pkg/sql/opt/constraint/constraint_set.go
@@ -289,6 +289,29 @@ func (s *Set) ExtractConstCols(evalCtx *tree.EvalContext) opt.ColSet {
 	return res
 }
 
+// ExtractValueForConstCol extracts the value for a constant column returned
+// by ExtractConstCols.
+func (s *Set) ExtractValueForConstCol(evalCtx *tree.EvalContext, col opt.ColumnID) tree.Datum {
+	if s == Unconstrained || s == Contradiction {
+		return nil
+	}
+	for i := 0; i < s.Length(); i++ {
+		c := s.Constraint(i)
+		colOrd := -1
+		for j := 0; j < c.Columns.Count(); j++ {
+			if c.Columns.Get(j).ID() == col {
+				colOrd = j
+				break
+			}
+		}
+		// The column must be part of the constraint's "exact prefix".
+		if colOrd != -1 && c.ExactPrefix(evalCtx) > colOrd {
+			return c.Spans.Get(0).StartKey().Value(colOrd)
+		}
+	}
+	return nil
+}
+
 // IsSingleColumnConstValue returns true if the Set contains a single constraint
 // on a single column which allows for a single constant value. On success,
 // returns the column and the constant value.

--- a/pkg/sql/opt/constraint/constraint_set_test.go
+++ b/pkg/sql/opt/constraint/constraint_set_test.go
@@ -270,53 +270,52 @@ func TestExtractCols(t *testing.T) {
 }
 
 func TestExtractConstCols(t *testing.T) {
+	type vals map[opt.ColumnID]string
 	type testCase struct {
 		constraints []string
-		expected    opt.ColSet
+		expected    vals
 	}
 
-	cols := opt.MakeColSet
-
 	cases := []testCase{
-		{[]string{`/1: [/10 - /10]`}, cols(1)},
-		{[]string{`/-1: [/10 - /10]`}, cols(1)},
-		{[]string{`/1: [/10 - /11]`}, cols()},
-		{[]string{`/1: [/10 - ]`}, cols()},
-		{[]string{`/1/2: [/10/2 - /10/4]`}, cols(1)},
-		{[]string{`/1/-2: [/10/4 - /10/2]`}, cols(1)},
-		{[]string{`/1/2: [/10/2 - /10/2]`}, cols(1, 2)},
-		{[]string{`/1/2: [/10/2 - /12/2]`}, cols()},
-		{[]string{`/1/2: [/9/2 - /9/2] [/10/2 - /12/2]`}, cols()},
-		{[]string{`/1: [/10 - /10] [/12 - /12]`}, cols()},
+		{[]string{`/1: [/10 - /10]`}, vals{1: "10"}},
+		{[]string{`/-1: [/10 - /10]`}, vals{1: "10"}},
+		{[]string{`/1: [/10 - /11]`}, vals{}},
+		{[]string{`/1: [/10 - ]`}, vals{}},
+		{[]string{`/1/2: [/10/2 - /10/4]`}, vals{1: "10"}},
+		{[]string{`/1/-2: [/10/4 - /10/2]`}, vals{1: "10"}},
+		{[]string{`/1/2: [/10/2 - /10/2]`}, vals{1: "10", 2: "2"}},
+		{[]string{`/1/2: [/10/2 - /12/2]`}, vals{}},
+		{[]string{`/1/2: [/9/2 - /9/2] [/10/2 - /12/2]`}, vals{}},
+		{[]string{`/1: [/10 - /10] [/12 - /12]`}, vals{}},
 		{
 			[]string{
 				`/1: [/10 - /10]`,
 				`/2: [/8 - /8]`,
 				`/-3: [/13 - /7]`,
 			},
-			cols(1, 2),
+			vals{1: "10", 2: "8"},
 		},
 		{
 			[]string{
 				`/1/2: [/10/4 - /10/5] [/12/4 - /12/5]`,
 				`/2: [/4 - /4]`,
 			},
-			cols(2),
+			vals{2: "4"},
 		},
-		{[]string{`/1: [/10 - /11)`}, cols()},
+		{[]string{`/1: [/10 - /11)`}, vals{}},
 		// TODO(justin): column 1 here is constant but we don't infer it as such.
 		{
 			[]string{
 				`/2/1: [/900/4 - /900/4] [/1000/4 - /1000/4] [/1100/4 - /1100/4] [/1400/4 - /1400/4] [/1500/4 - /1500/4]`,
 			},
-			cols(),
+			vals{},
 		},
 		{
 			[]string{
 				`/1: [/2 - /3]`,
 				`/2/1: [/10/3 - /11/1]`,
 			},
-			cols(),
+			vals{},
 		},
 	}
 
@@ -328,9 +327,23 @@ func TestExtractConstCols(t *testing.T) {
 			cs = cs.Intersect(evalCtx, SingleConstraint(&constraint))
 		}
 		cols := cs.ExtractConstCols(evalCtx)
-		if !tc.expected.Equals(cols) {
-			t.Errorf("expected constant columns from %s to be %s, was %s", cs, tc.expected, cols)
+		var expCols opt.ColSet
+		for col := range tc.expected {
+			expCols.Add(col)
 		}
+		if !expCols.Equals(cols) {
+			t.Errorf("%s: expected constant columns be %s, was %s", cs, expCols, cols)
+		}
+		cols.ForEach(func(col opt.ColumnID) {
+			val := cs.ExtractValueForConstCol(evalCtx, col)
+			if val == nil {
+				t.Errorf("%s: no const value for column %d", cs, col)
+				return
+			}
+			if actual, expected := val.String(), tc.expected[col]; actual != expected {
+				t.Errorf("%s: expected value %s for column %d, got %s", cs, expected, col, actual)
+			}
+		})
 	}
 }
 

--- a/pkg/sql/opt/memo/extract.go
+++ b/pkg/sql/opt/memo/extract.go
@@ -294,3 +294,20 @@ func ExtractConstColumns(
 	}
 	return fixedCols
 }
+
+// ExtractValueForConstColumn returns the constant value of a column returned by
+// ExtractConstColumns.
+func ExtractValueForConstColumn(
+	on FiltersExpr, mem *Memo, evalCtx *tree.EvalContext, col opt.ColumnID,
+) tree.Datum {
+	for i := range on {
+		scalar := on[i]
+		scalarProps := scalar.ScalarProps()
+		if scalarProps.Constraints != nil {
+			if val := scalarProps.Constraints.ExtractValueForConstCol(evalCtx, col); val != nil {
+				return val
+			}
+		}
+	}
+	return nil
+}

--- a/pkg/sql/opt/xform/custom_funcs.go
+++ b/pkg/sql/opt/xform/custom_funcs.go
@@ -1777,8 +1777,8 @@ func (c *CustomFuncs) fixedColsForZigzag(
 ) (fixedCols opt.ColList, vals memo.ScalarListExpr, typs []types.T) {
 	for i, cnt := 0, index.ColumnCount(); i < cnt; i++ {
 		colID := tabID.ColumnID(index.Column(i).Ordinal)
-		val, _, ok := c.findConstantFilter(filters, colID)
-		if !ok {
+		val := memo.ExtractValueForConstColumn(filters, c.e.mem, c.e.evalCtx, colID)
+		if val == nil {
 			break
 		}
 		if vals == nil {
@@ -1944,6 +1944,10 @@ func (c *CustomFuncs) GenerateZigzagJoins(
 			rightFixedCols, rightVals, rightTypes := c.fixedColsForZigzag(
 				iter2.index, scanPrivate.Table, filters,
 			)
+
+			if len(leftFixedCols) != leftFixed.Len() || len(rightFixedCols) != rightFixed.Len() {
+				panic(errors.AssertionFailedf("could not populate all fixed columns for zig zag join"))
+			}
 
 			zigzagJoin.LeftFixedCols = leftFixedCols
 			zigzagJoin.RightFixedCols = rightFixedCols

--- a/pkg/sql/opt/xform/testdata/rules/join
+++ b/pkg/sql/opt/xform/testdata/rules/join
@@ -1756,6 +1756,20 @@ memo (optimized, ~13KB, required=[presentation: q:2,r:3])
  ├── G16: (variable r)
  └── G17: (const 2)
 
+# Case where the fixed columns are extracted from a complicated expression.
+opt
+SELECT q,r FROM pqr WHERE q = 1 AND ((r < 1 AND r > 1) OR (r >= 2 AND r <= 2))
+----
+inner-join (zigzag pqr@q pqr@r)
+ ├── columns: q:2!null r:3!null
+ ├── eq columns: [1] = [1]
+ ├── left fixed columns: [2] = [1]
+ ├── right fixed columns: [3] = [2]
+ ├── fd: ()-->(2,3)
+ └── filters
+      ├── q:2 = 1 [outer=(2), constraints=(/2: [/1 - /1]; tight), fd=()-->(2)]
+      └── ((r:3 < 1) AND (r:3 > 1)) OR ((r:3 >= 2) AND (r:3 <= 2)) [outer=(3), constraints=(/3: [/2 - /2]), fd=()-->(3)]
+
 # Nested zigzag case - zigzag join needs to be wrapped in a lookup join to
 # satisfy required columns.
 opt

